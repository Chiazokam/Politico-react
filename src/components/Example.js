import React from "react";
import { connect } from 'react-redux';
import { increment, decrement } from '../actions/example';

class Example extends React.Component {

  increment = () => {
    this.props.increment();
  };

  decrement = () => {
    this.props.decrement();
  };

  render() {
    return (
      <div className="counter">
        <h2>Counter</h2>
        <div>
          <button onClick={this.decrement}>-</button>
          <span className="count">{this.props.count}</span>
          <button onClick={this.increment}>+</button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => { 
  return {
    count: state.example.count };
  }

const mapDispatchToProps = {
  increment,
  decrement,
};
    
export default connect(mapStateToProps, mapDispatchToProps)(Example);
