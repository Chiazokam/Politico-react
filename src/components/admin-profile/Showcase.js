import React, { Component } from 'react';

class Showcase extends Component {
  render() {
    return (
      <div id="main-profile">
        <div className="flex-container">
          <div className="card">
              <a href="/parties/new"><button type="button" className="btn btn-trans">Create a Political Party</button></a>
          </div>  
          <div className="card">
              <a href="/parties"><button type="button" className="btn btn-trans">View Political Parties</button></a>
          </div> 
          <div className="card">
              <a href="/offices/new"><button type="button" className="btn btn-trans">Create a Political Office</button></a>
          </div> 
          <div className="card">
              <a href="#"><button type="button" className="btn btn-trans">View Political Offices</button></a>
          </div> 
        </div>
      </div>
    )
  }
}

export default Showcase;
