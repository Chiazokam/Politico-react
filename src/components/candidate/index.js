import { BecomeCandidate } from './BecomeCandidate';
import { CreateCandidate } from './CreateCandidate'

export { BecomeCandidate, CreateCandidate };
