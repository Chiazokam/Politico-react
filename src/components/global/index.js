import Button from './Button';
import Footer from './Footer';
import NavList from './NavList';
import Header from './Header';
import TextStrip from './TextStrip';
import PlainStrip from './PlainStrip';
import Container from './Container';
import FormField from './FormField';
import Url from './helper';
import Logout from './Logout';
import PartyCard from './Party-card';
import OfficeCard from './Office-card';
import ToastMessage from './ToastMessage';

export {
  Header,
  Button,
  Footer,
  NavList,
  TextStrip,
  PlainStrip,
  Container,
  FormField,
  Url,
  Logout,
  PartyCard,
  OfficeCard,
  ToastMessage
};
