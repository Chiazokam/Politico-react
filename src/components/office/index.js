import CreateOffice from './Create-office';

export { CreateOffice };
