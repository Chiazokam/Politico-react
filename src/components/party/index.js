import CreateParty from './create-party';

export { CreateParty };
