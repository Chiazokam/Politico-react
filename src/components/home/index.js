import Header from '../global/Header';
import Showcase from './showcase';

export { Header, Showcase };
