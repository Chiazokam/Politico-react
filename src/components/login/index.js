import Showcase from './Showcase';
import ResetPasswordModal from './Reset-password-modal';

export { Showcase, ResetPasswordModal };
