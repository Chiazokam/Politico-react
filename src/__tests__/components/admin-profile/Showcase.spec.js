import React from 'react';
import { shallow } from 'enzyme';
import { Showcase } from '../../../components/admin-profile';

describe('Admin Profile', () => {
  test('should render Admin Profile component correctly', () => {
    const wrapper = shallow(<Showcase />);

    expect(wrapper).toMatchSnapshot();
  });
});
