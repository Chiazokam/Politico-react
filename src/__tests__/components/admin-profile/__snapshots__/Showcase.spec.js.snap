// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Profile should render Admin Profile component correctly 1`] = `ShallowWrapper {}`;
